### PR TITLE
Added support for building APS packages [Issue #1034]

### PR DESCRIPTION
**Pull request by SalmanPK**
http://ocportal.com/tracker/view.php?id=1034
These two are mentioned in the issue tracker are not actually needed for Gold level AFAIK but let me know  if I should add support for them:
Backup process implementation, to achieve Gold level
Certificate signing implementation, to achieve Gold level

TODO: make_release.php has to link through to certification form and say what to fill, make it super simple

### CODE REVIEW

File: sources_custom/make_release.php [hunk 1]
```diff
@@ -72,7 +73,7 @@ function make_installers($skip_file_grab = false)
         $out .= '</ul>';
 
         make_files_manifest();
-        make_database_manifest();
+        //make_database_manifest();
```
Comment (SalmanPK) [inline sources_custom/make_release.php]: Whoops!

**Comment by SalmanPK** [comment]
The certification process:
aps_package_certification_guide_2_1-42

First we have to be granted access to the APS Certification System. Emailed them yesterday, no response yet.

**Comment by chrisgraham** [comment]
Nice work with the XML DOM stuff.

"Whoops!" - include a fix of your accidental line comment in your commit. Don't make me do it ;-)

Merge in yourself when ready. I'd rather not merge the pull myself, laziness, and I like to make quick changes without messing too much with branches - especially during this v10 alpha stage.

PSR-2:
Your sources_custom/make_release.php has mixtures of tabs and spaces. It has to be 4-space indentation.
&
if requires braces for it:
if ($success === false) warn_exit('Failed to build APP-LIST.xml');
--->
If you look in .editorconfig you'll see some PHPStorm settings are commented into there. So just apply that in PHPStorm and do an auto-format. Then check the diff to make sure it hasn't changed pre-existing code (it shouldn't, as I auto-formatted the whole project with these settings).

// TODO for install.sql: Cache tables should be empty
You can remove this comment. It should not matter. Composr auto-detects a changed base URL from the DB and auto-flushes out caches.

Put it to 64MB minimum memory

Include xml and mbstring and curl extensions in extensions list

PHP min is 5.1

MySQL min is 5.5.3

Still need to do release link in adminzone/pages/minimodules_custom/make_release.php
Not heard back from them yet I guess? At least put in a TODO in this file and include "#651" in that TODO (by convention we ensure all TODOs in the code are linked to a tracker issue, else CQC will trigger a warning in our test set).

adminzone/pages/minimodules_custom/_make_release.php needs modifying to auto-create download entry on compo.sr for the uploaded built file. We already do for installatron etc.

Change prefix from cms10_ to cms_
We only included version in debug mode installs. On live it's a huge PITA.

You can strip:
$SITE_INFO['db_forums']='@@DB_MAIN_NAME@@';
$SITE_INFO['db_forums_host']='@@DB_MAIN_HOST@@:@@DB_MAIN_PORT@@';
$SITE_INFO['db_forums_user']='@@DB_MAIN_LOGIN@@';
$SITE_INFO['db_forums_password']='@@DB_MAIN_PASSWORD@@';
$SITE_INFO['cns_table_prefix']='cms10_';
...as they'll just auto-detect from the site config anyway. Let's keep it cleaner for these installs.

You need to add:
$SITE_INFO['multi_lang_content'] = '0';
$SITE_INFO['session_cookie'] = 'cms_session__<somerandomalphanumericstring>';
where it really is randomish, e.g. uniqid('')
This will segregate cookies in v10. As you put the path as (/) good, but we want to allow multi installs on same domain

If backup/certificate stuff is not needed, skip it. I probably just got this requirement from you anyway.

I am worried about keeping chmod rules consistent. There are so many places these are set now. I'm going to work on an integration test to ensure things stay consistent.

**Comment by chrisgraham** [comment]
Ok, the new chmod consistency test is added:
https://github.com/ocproducts/composr/blob/master/_tests/tests/unit_tests/chmod_consistency.php
I found a tonne of problems in the existing files, and fixed them. Please make sure it now passes for yours.

**Comment by SalmanPK** [comment]
>// TODO for install.sql: Cache tables should be empty
>You can remove this comment. It should not matter. Composr auto-detects a changed base URL from >the DB and auto-flushes out caches.

I added that TODO because I was getting links for images, styleseheets etc. hard-coded to the local IP 192.168.1.100, when installing the APS package on a completely different host. So there's probably a bug then which needs fixing.

**Comment by SalmanPK** [comment]
>Change prefix from cms10_ to cms_
>We only included version in debug mode installs. On live it's a huge PITA.
Can you update the install.sql database dump for this first? This will break the APS package otherwise.. I want to be able to test it.

**Comment by SalmanPK** [comment]
>$SITE_INFO['session_cookie'] = 'cms_session__';
where it really is randomish, e.g. uniqid('')
This will segregate cookies in v10. As you put the path as (/) good, but we want to allow multi installs on same domain

So it should be `$SITE_INFO['session_cookie'] = 'cms_session__' . uniqid('');`?

**Comment by SalmanPK** [comment]
Aw man I am starting to hate `git pull` now. There are errors everywhere.. so frustrating. There go a few hours of lost productivity again.. shoulda just done it for v9.

**Pull request closed without merging**